### ISSUE TITLE
[Feature] Specify table API which Rest catalog needs support

### ISSUE THREAD

**Issue by jerry-024** [apache/paimon#4715]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Motivation

Specify table API which Rest catalog needs support.

### Solution
I have searched other catalog APIs like iceberg Rest catalog and unity catalog, 
- the APIs we need support with are
  - getTable
  - listTables
  - dropTable
  - createTable
  - renameTable
  - alterTable
  - ~~getTableLocation~~： as it only use when clone table
- need discuss
  - tableExists: as we needn't use this API in common so getTable is enough

##### iceberg
- listTables
- createTable
- tableExists
- dropTable
- renameTable
- loadTable
- invalidateTable 
- registerTable

##### unity catalog
- listTables
- getTable
- deleteTable
- tableExists

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by adrian-wang** [comment]
How about `getTables` to batch get tables to avoid network communication overhead?

**Comment by jerry-024** [comment]
What's the different between `listTables` and `getTables`?

**Comment by JingsongLi** [comment]
> Are there any plans to support the `snapshot`, `branch`, or `tag` API ?

Good point, these need some refactor, we will discuss them in follow issues.

**Comment by JingsongLi** [comment]
> How about `getTables` to batch get tables to avoid network communication overhead?

@adrian-wang Do you know which scenarios need this batch API? I cannot see it in computation engines.

**Comment by sundapeng** [comment]
> > Are there any plans to support the `snapshot`, `branch`, or `tag` API ?
> 
> Good point, these need some refactor, we will discuss them in follow issues.

I'm looking forward to the discussion.

**Comment by sundapeng** [comment]
> > How about `getTables` to batch get tables to avoid network communication overhead?
> 
> @adrian-wang Do you know which scenarios need this batch API? I cannot see it in computation engines.

@JingsongLi , APIs for tables can optimize certain management scenarios, such as `listTables` typically returning table names and `getTables` needing to return full table objects. Multi-table join scenarios may also benefit.

Based on my previous experience, batch APIs for partitions can also significantly improve performance in some queries.

Any thought? @adrian-wang

**Comment by jerry-024** [comment]
1. Maybe we could follow unity catalog like:
`listTableSummaries ` -> only return table name
`listTables` -> return full table objects
2. if we use `getTables ` to return full table objects, we need to support pagination in this API.

**Comment by adrian-wang** [comment]
@sundapeng +1
